crypto/pkcs11/pkcs11helpers.go

```diff
@@ -228,6 +232,9 @@ func pkcs11UriLogin(p11uri *pkcs11uri.Pkcs11URI) (ctx *pkcs11.Ctx, session pkcs1
 		if err != nil {
 			return nil, 0, errors.Wrapf(err, "OpenSession to slot %d failed", slotid)
 		}
+		if len(pin) == 0 {
```

Comment:
Hmm, can we refactor the slot ID handling as a function, repeated code with different handling styles in both branches. Maybe just an if statement for 

`if slotid>=0 { slots = []uint{uint(slotid)}} else ...`

Reply:
Yes, better. Done.